patches/tock/12-pending-deadlock.patch

```diff
@@ -0,0 +1,79 @@
+diff --git a/capsules/src/usb/usbc_ctap_hid.rs b/capsules/src/usb/usbc_ctap_hid.rs
+index 16b80cb10..c71ed639a 100644
+--- a/capsules/src/usb/usbc_ctap_hid.rs
++++ b/capsules/src/usb/usbc_ctap_hid.rs
+@@ -120,7 +120,6 @@ struct EndpointState {
+ 
+     tx_packet: OptionalCell<[u8; 64]>,
+     pending_in: Cell<bool>,
+-    pending_out: Cell<bool>,
+     // Is there a delayed packet?
+     delayed_out: Cell<bool>,
+ }
+@@ -133,7 +132,6 @@ impl EndpointState {
+             out_buffer: Buffer64::default(),
+             tx_packet: OptionalCell::empty(),
+             pending_in: Cell::new(false),
+-            pending_out: Cell::new(false),
+             delayed_out: Cell::new(false),
+         }
+     }
+@@ -142,6 +140,9 @@ impl EndpointState {
+ pub struct ClientCtapHID<'a, 'b, C: 'a> {
+     client_ctrl: ClientCtrl<'a, 'static, C>,
+ 
++    // Is there a pending OUT transaction happening?
++    pending_out: Cell<bool>,
++
+     endpoints: [EndpointState; NUM_ENDPOINTS],
+ 
+     // Interaction with the client
+@@ -264,6 +265,7 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+                 LANGUAGES,
+                 strings,
+             ),
++            pending_out: Cell::new(false),
+             endpoints: [
+                 EndpointState::new(ENDPOINT_NUM),
+                 #[cfg(feature = "vendor_hid")]
+@@ -306,12 +308,14 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+     }
+ 
+     pub fn receive_packet(&'a self, app: &mut App) {
+-        for (_, s) in self.endpoints.iter().enumerate() {
+-            if s.pending_out.get() {
+-                // The previous packet has not yet been received, reject the new one.
+-                continue;
+-            } else {
+-                s.pending_out.set(true);
++        if self.pending_out.get() {
++            // The previous packet has not yet been received, reject the new one.
++        } else {
++            self.pending_out.set(true);
++            // Process the first endpoint that has a delayed packet.
++            // TODO(liamjm): Determine if a round-robin approach reduces latency
++            // in the worst case.
```

Comment:
This is only an issue in case of long-lasting contention right? I think it's ok if the behavior is unfair. We could however decide which endpoint should be prioritized (by adding a `.rev()` after `.enumerate()`). Maybe we want to prefer the second page. That would be specific to OpenSK, but this driver is already specific to OpenSK.

Reply:
Agreed. I"ll do this in the subsequent PR.